### DISCUSSION TITLE
SpikeGLX in CellExplorer

### DISCUSSION THREAD

**Discussion by GuilleSGCUPO** [petersenpeter/CellExplorer#77] [Q&A] [answered]
Hello again :), I was trying to run the 'CellExplorer_NeuropixelsTutorial.m' but I realized the '.dat' archive it asked for, is not the kind of archive SpikeGLX provides (.ap.bin):

![image](https://user-images.githubusercontent.com/108993072/197579886-170ba0b8-a8a5-4629-9ea7-d92e52de8b38.png)

Is there any way to upload datasets from SpikeGLX software?
Thank you in advance,
Guille :)

**Comment by petersenpeter** [comment, marked as answer]
Hi Guille
The file formats are the same, just different extensions. You can either rename the file, e.g. basename.dat, or use the existing file and specify the full name in the session struct (can also be done in the session GUI):
```m
session.extracellular.fileName = 'basename.ap.bin';
```

**Reply by GuilleSGCUPO** [reply]
Thank you Peter for both answers. For the first one I cloned the code instead of download it and now it works. For the second one, by renaming the archives I was able to run them on CellExplorer. However, I have had new issues, I hope you may help whenever you can:

 - When I build the "session" struct, it reads "rez*.mat file does not exist", but I have the rez files on the basepath, although it is able to find the Phy data and run CellExplorer. 
 
 - When it imports the clusters found in the Phy archives, they correspond to the clusters we manually curated in Kilosort's Gui. Nevertheless, the waveform (raw and averages), for every cluster, does not relate to neuronal spikes (it seems to be noise) nor the waveforms we saw in Kilosort's Gui. I attached a screenshort showing the same Cluster on both softwares so you have an idea:

 
![image](https://user-images.githubusercontent.com/108993072/198341037-2441f1dd-3f9a-4548-aea5-f0f48a71ef96.png)


- Following the previous topic, when the session summary figure appears, the depth graph does not relate to the actual depth of the several recording channels where clusters were found (probe shown on the right):


![image](https://user-images.githubusercontent.com/108993072/198343230-d734ac81-7d0c-4aa4-8b36-13c8c2c6e5e2.png)


Do you have any ideas about what the problem may be? And if you have any additional comments or suggestions we would be pleased to receive them. Thanks by advance and we wait for your answer.

**Comment by petersenpeter** [comment]
Hey. Happy to help. First, check the display called "Waveforms (across channels)". It is a similar plot as shown in your first screenshot from Phy (the WaveformView). It should show the same waveforms and why you have this weird waveform. 
It most likely happened because you had noisy channels from which the largest waveform was detected. 

You can also share the cell metrics with me, as it makes it easier to troubleshoot.
